services/cdn/wait/wait.go

```diff
@@ -39,17 +40,21 @@ func CreateDistributionPoolWaitHandler(ctx context.Context, api APIClientInterfa
 		}
 		if *distribution.Distribution.Id == distributionId {
 			switch *distribution.Distribution.Status {
-			case DistributionStateActive:
-				return true, distribution, err
+			case DistributionStatusActive:
+				return true, distribution, nil
+			case DistributionStatusCreating, DistributionStatusUpdating:
+				return false, nil, nil
+			case DistributionStatusDeleting:
+				return true, nil, fmt.Errorf("creating CDN distribution failed")
```

Comment:
The original error should be contained within the returned error to facilitate debugging:

```suggestion
				return true, nil, fmt.Errorf("creating CDN distribution failed: %w",err)
```
This affects the following statements as well

Reply:
there was no error here, `err` is nil...